pkg/bitcoin/electrum/electrum_integration_test.go

```diff
@@ -1,119 +1,156 @@
-			missingTransactionInBlock: "errNo: 0, errMsg: tx not found or is unconfirmed",
-		},
-	},
-	"electrumx tcp": {
-		clientConfig: Config{
-			URL:                 "tcp://tn.not.fyi:55001",
-			RequestTimeout:      timeout,
-			RequestRetryTimeout: timeout * 2,
-		},
-		errorMessages: expectedErrorMessages{
-			missingTransaction:        "errNo: 2, errMsg: daemon error: DaemonError({'code': -5, 'message': 'No such mempool or blockchain transaction. Use gettransaction for wallet transactions.'})",
-			missingBlockHeader:        "errNo: 1, errMsg: height 4,294,967,295 out of range",
-			missingTransactionInBlock: "errNo: 1, errMsg: tx aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa not in block at height 123,456",
+			RequestRetryTimeout: timeout * 6,
 		},
+		network: bitcoin.Testnet,
 	},
 	"electrumx wss": {
-		clientConfig: Config{
+		clientConfig: electrum.Config{
 			URL:                 "wss://electrumx-server.test.tbtc.network:8443",
 			RequestTimeout:      timeout,
 			RequestRetryTimeout: timeout * 2,
 		},
-		errorMessages: expectedErrorMessages{
-			missingTransaction:        "errNo: 2, errMsg: daemon error: DaemonError({'code': -5, 'message': 'No such mempool or blockchain transaction. Use gettransaction for wallet transactions.'})",
-			missingBlockHeader:        "errNo: 1, errMsg: height 4,294,967,295 out of range",
-			missingTransactionInBlock: "errNo: 1, errMsg: tx aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa not in block at height 123,456",
-		},
+		network: bitcoin.Testnet,
 	},
 	"fulcrum tcp": {
-		clientConfig: Config{
-			URL:                 "tcp://testnet.aranguren.org:51001",
+		clientConfig: electrum.Config{
+			URL:                 "tcp://blackie.c3-soft.com:57005",
 			RequestTimeout:      timeout,
 			RequestRetryTimeout: timeout * 2,
 		},
-		errorMessages: expectedErrorMessages{
-			missingTransaction:        "errNo: 2, errMsg: daemon error: DaemonError({'code': -5, 'message': 'No such mempool or blockchain transaction. Use gettransaction for wallet transactions.'})",
-			missingBlockHeader:        "errNo: 1, errMsg: Invalid height",
-			missingTransactionInBlock: "errNo: 1, errMsg: No transaction matching the requested hash found at height 123456",
-		},
+		network: bitcoin.Testnet,
 	},
 }
 
+var invalidTxID bitcoin.Hash
+
+//go:linkname readEmbeddedServers github.com/keep-network/keep-core/config.readElectrumUrls
+func readEmbeddedServers(network bitcoin.Network) ([]string, error)
+
+func init() {
+	var err error
+
+	readServers := func(network bitcoin.Network) error {
+		servers, err := readEmbeddedServers(network)
+		if err != nil {
+			return err
+		}
+
+		for _, server := range servers {
+			serverName := fmt.Sprintf("embedded/%s/%s", network.String(), server)
+			testConfigs[serverName] = testConfig{
+				clientConfig: electrum.Config{
+					URL:                 server,
+					RequestTimeout:      timeout * 2,
+					RequestRetryTimeout: timeout * 4,
```

Comment:
I am confused why we use x2/x4 multipliers for embedded servers but x1/x2 for testnet fulcrum and testnet electrumx from `var testConfigs`. Shouldn't we assign all of them the same multipliers? Blockstream server is an obvious exception given #3612.

Reply:
It was to cover Blockstream servers, as they required longer timeouts. I'll modify it, since we removed blockstream.

Reply:
4298722b6081c9a6f717b1fb5a41dd47c0cbbef1